metricflow/api/simple_api.py

```diff
@@ -0,0 +1,259 @@
+            metric_names: Names of metrics to get common dimensions from.
+
+        Returns:
+            A list of Dimension objects containing metadata.
+        """
+        return self.mf_engine.simple_dimensions_for_metrics(metric_names=metric_names)
+
+    def list_materializations(self) -> List[Materialization]:
+        """Retrieves a list of materialization names.
+
+        Returns:
+            A list of Materialization objects containing metadata.
+        """
+        return self.mf_engine.list_materializations()
+
+    def get_dimension_values(
+        self, metric_name: str, dimension_name: str, start_time: Optional[str] = None, end_time: Optional[str] = None
+    ) -> List[str]:
+        """Retrieves a list of dimension values given a [metric_name, dimension_name].
+
+        Args:
+            metric_name: Name of metric that contains the group_by.
+            dimension_name: Name of group_by to get values from.
+            start_time: Get data for the start of this time range.
+            end_time: Get data for the end of this time range.
+
+        Returns:
+            A list of dimension values as string.
+        """
+        parsed_start_time = convert_to_datetime(start_time)
+        parsed_end_time = convert_to_datetime(end_time)
+        return self.mf_engine.get_dimension_values(
+            metric_name=metric_name,
+            get_group_by_values=dimension_name,
+            time_constraint_start=parsed_start_time,
+            time_constraint_end=parsed_end_time,
+        )
+
+    def materialize(
+        self, materialization_name: str, start_time: Optional[str] = None, end_time: Optional[str] = None
+    ) -> SqlTable:
+        """Builds a table containing metrics and dimensions from a materialization definition.
+
+        This can be very expensive if a large time range is provided.
+
+        Args:
+            materialization_name: Name of materialization
+            start_time: Materialized for the start of this time range.
+            end_time: Materialized for the end of this time range.
+
+        Returns:
+            SqlTable object of the materialized table.
+        """
+        parsed_start_time = convert_to_datetime(start_time)
+        parsed_end_time = convert_to_datetime(end_time)
+        return self.mf_engine.materialize(
+            materialization_name=materialization_name,
+            time_constraint_start=parsed_start_time,
+            time_constraint_end=parsed_end_time,
+        )
+
+    def drop_materialization(self, materialization_name: str) -> bool:
+        """Drops the table associated with a materialization definition.
+
+        Args:
+            materialization_name: Name of materialization to drop.
+
+        Returns:
+            True if a table has been drop, False if table doesn't exist.
+        """
+        return self.mf_engine.drop_materialization(materialization_name=materialization_name)
+
+    def validate_configs(self) -> List[ValidationIssueType]:
```

Comment:
Do we want `Tuple[ValidationIssueType, ...]` here instead? I think it might be better.

1. It simplifies a number of things, starting with the body of this method
2. I don't think we want people appending to the set of returned issues - if they have custom validation wrappers they can always do `mf.validate_configs + custom_mf.validate_configs` or similar.

Reply:
changed typing back to Tuple